swift/Sources/FlatBuffers/ByteBuffer.swift

```diff
@@ -352,7 +358,14 @@ public struct ByteBuffer {
   ///   - position: the index of the object in the buffer
   @inline(__always)
   public func read<T>(def: T.Type, position: Int) -> T {
-    _storage.memory.advanced(by: position).load(as: T.self)
+    #if swift(>=5.7)
+    if readUnalignedBuffers {
+      return _storage.memory.advanced(by: position).loadUnaligned(as: T.self)
+    }
+    return _storage.memory.advanced(by: position).load(as: T.self)
+    #else
+    return _storage.memory.advanced(by: position).load(as: T.self)
+    #endif
```

Comment:
There is no prettier way of doing this to support aligned and unaligned memory. So for now this approach would be the best, unless swift 5.9 and above would come with a nice solution that solves this for us without the copying.
PS: the benchmarks are done while using the new read method, which means in terms of writing buffers we didn't lose much in terms of performance.